babel/messages/pofile.py

```diff
@@ -73,6 +75,10 @@ def denormalize(string):
         return unescape(string)
 
 
+class PoFileError(Exception):
+    """Exception thrown by PoParser when an invalid po file is encountered."""
```

Comment:
It would be nice if this was a more full-fledged exception, to make it easier to use programmatically by client code:

```python
class PoFileError(Exception):
  def __init__(self, message, catalog, line, lineno):
    super(PoFileError, self).__init__('{message} on {lineno}'.format(message=message, lineno=lineno))
    self.catalog = catalog
    self.line = line
    self.lineno = lineno
```
and naturally the `raise` below would be
```python
raise PoFileError(msg, self.catalog, line, lineno)
```
This way client code can introspect the error without having to try to parse the message.

What do you think?

Reply:
Sounds like a good idea, will implement soon